src/rollback_to_stable/rts_btree.c

```diff
@@ -775,24 +786,10 @@ __rts_btree_abort_ondisk_kv(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
       upd->type == WT_UPDATE_TOMBSTONE ? "true" : "false",
       __wt_key_string(session, key->data, key->size, S2BT(session)->key_format, key_string));
 
-    /*
-     * It is possible for both RTS and the eviction thread to remove entries from the history store
-     * at the same time: When we evict a key with a tombstone with durable timestamp "none,"
-     * __wti_rec_hs_clear_on_tombstone also removes the corresponding entries from the history
-     * store, which is necessary for correctness. If the two threads try to remove the same history
-     * store key at around the same time, we could get a WT_NOTFOUND here.
-     *
-     * FIXME-WT-13198: Check if this race can be avoided, in which case we can remove the check for
-     * WT_NOTFOUND and treat it like other errors.
-     */
     if (rip != NULL)
-        WT_ERR_NOTFOUND_OK(__rts_btree_row_modify(session, ref, &upd, key), true);
-    else
-        WT_ERR_NOTFOUND_OK(__rts_btree_col_modify(session, ref, &upd, recno), true);
-    if (ret == WT_NOTFOUND && WT_IS_HS(session->dhandle))
-        ret = 0;
+        WT_ERR(__rts_btree_row_modify(session, ref, &upd, key));
```

Comment:
This error handling code is active only for the history store, It is fine to leave it as it is. Do you think leaving this code may silently ignore any issues that can occur with RTS and history store if we have that code?

Reply:
I'm wondering the same thing. We don't want to mask a potential logic error, so that we would be able to find the root cause as Chenhao did here, as that'd be far superior to this bandaid solution that I had here.

At the same time, it may also be okay leave it as is, as it would just keep ignoring the cases where we want to delete a record that doesn't exist, so the effect would be the same.

If there is an error returned here, could we make it fail at least in testing, or at least log a message that would make the test fail (working around that RTS tests currently ignore everything with WT_VERB_RTS)? I'd be happy to defer to y'all about which way we should go for production builds, as that involves a different set of tradeoffs. Just my 2c...

Reply:
I change this code back because it is specifically handling HS case and it is not a general solution.